File: src/main/java/com/jcloisterzone/action/GoldPieceAction.java
package com.jcloisterzone.action;

import com.jcloisterzone.board.Position;
import com.jcloisterzone.game.capability.GoldminesCapability.GoldToken;
import com.jcloisterzone.wsio.message.PlaceTokenMessage;
import com.jcloisterzone.wsio.message.WsInGameMessage;
import io.vavr.collection.Set;

public class GoldPieceAction extends SelectTileAction {

    public GoldPieceAction(Set<Position> options) {
        super(options);
    }

    @Override
    public WsInGameMessage select(Position pos) {
        return new PlaceTokenMessage(GoldToken.GOLD, pos);
    }

    @Override
    public String toString() {
        return "place gold piece";
    }
}

File: src/main/ts/com/jcloisterzone/action/GoldPieceAction.ts
import type { Set } from "../../../io/vavr/Set.js";
import type { Position } from "../board/Position.js";
import { SelectTileAction } from "./SelectTileAction.js";

/** Offer to place the second gold piece on one of the tiles around a goldmine. */
export class GoldPieceAction extends SelectTileAction {
  static readonly simpleName = "GoldPieceAction";

  constructor(options: Set<Position>) {
    super(options);
  }
}
